File: src/protocol/BlindsDOWNBuilder.cpp
#include "BlindsDOWNBuilder.h"
#include "BlindsDOWNCommand.h"

using namespace std;
using namespace communication;

BlindsDOWNBuilder::BlindsDOWNBuilder()
{}

BlindsDOWNBuilder::~BlindsDOWNBuilder()
{}

unique_ptr<Command> BlindsDOWNBuilder::create(const vector<uint8_t> &commandInBytes)
{
    auto command = make_unique<BlindsDOWNCommand>();

    return move(command);
}

File: mobile_client/app/src/main/java/protocol/BlindsDOWNBuilder.java
package protocol;

import java.io.*;
import java.util.ArrayList;

public class BlindsDOWNBuilder implements CommandBuilder, Closeable
	{
		public BlindsDOWNBuilder()
		{
		}
		public final void close()
		{
		}

		@Override
		public Command create(ArrayList<Integer> commandInBytes)
		{
			Command command  = new BlindsDOWNCommand();
			return command;
		}
	}